Mathlib/Algebra/Category/GroupCat/Injective.lean

```diff
@@ -72,72 +72,157 @@ instance injective_ratCircle : Injective <| of <| ULift.{u} <| AddCircle (1 : 
   have : Fact ((0 : ℚ) < 1) := ⟨by norm_num⟩
   injective_of_divisible _
 
-namespace enough_injectives_aux_proofs
+end AddCommGroupCat
 
-variable (A_ : AddCommGroupCat.{u})
+section injective_character
 
-/-- The next term of `A`'s injective resolution is `∏_{A →+ ℚ/ℤ}, ℚ/ℤ`. -/
-def next : AddCommGroupCat.{u} := of <|
-  (A_ ⟶ of <| ULift.{u} <| AddCircle (1 : ℚ)) → ULift.{u} (AddCircle (1 : ℚ))
+variable (R : Type u) [CommRing R]
+variable (M : Type w) [AddCommGroup M] [Module R M]
+variable (N : Type w') [AddCommGroup N] [Module R N]
 
-instance : Injective <| next A_ :=
+/--
+If `M` is an abelian group, its character module is defined to be the `Hom_ℤ(M, ℚ/ℤ)`
+-/
+def CharacterModule : Type w := M →ₗ[ℤ] (ULift.{w} <| AddCircle (1 : ℚ))
```

Comment:
I think this should be defined in another file and we should move everything not involving category theory there.